### ISSUE TITLE
[Bug]: Incorrect rotation shown in "Edit Image"

### ISSUE THREAD

**Issue by mnalis** [commons-app/apps-android-commons#5711]
### Summary

After clicking "Edit Image", picture (always?) seems to be in incorrect orientation (rotated 90 degrees counter-clockwise).

There is related issue https://github.com/commons-app/apps-android-commons/issues/5502, but I'm not sure if that applies?

### Steps to reproduce

1. share picture to Commons app (or in Commons app choose to add picture from gallery)
2. notice the picture is correctly oriented on main view (in `Step 1 of 4: Media Details` page)
3. click `Edit Image`

### Expected behaviour

expect that after clicking `Edit Image`, image will be shown in the same (correct) orientation that is was shown in step `2.` above

### Actual behaviour

image is shown rotated 90 degrees CCW.

### Device name

Samsung Galaxy S23+

### Android version

Android 14 (OneUI 6.1)

### Commons app version

5.0.1~af028cbdd (latest F-droid)

### Device logs

```
2024-05-04 21:34:51,674     [DEBUG] [file-logging-thread-1] [UploadMediaDetailAdapter$ViewHolder] : Description language code is: hr
2024-05-04 21:38:13,568     [DEBUG] [file-logging-thread-1] [EditActivity] : Rotation 0
2024-05-04 21:38:13,569     [DEBUG] [file-logging-thread-1] [EditActivity] : new Rotation 90
2024-05-04 21:39:26,319     [DEBUG] [file-logging-thread-1] [LocationServiceManager] : on location changed
2024-05-04 21:39:28,315     [DEBUG] [file-logging-thread-1] [LocationServiceManager] : on location changed
2024-05-04 21:39:29,322     [DEBUG] [file-logging-thread-1] [LocationServiceManager] : on location changed
2024-05-04 21:39:30,316     [DEBUG] [file-logging-thread-1] [LocationServiceManager] : on location changed
2024-05-04 21:39:32,313     [DEBUG] [file-logging-thread-1] [LocationServiceManager] : on location changed
2024-05-04 21:40:24,317     [DEBUG] [file-logging-thread-1] [LocationServiceManager] : on location changed

```

### Screen-shots

https://github.com/commons-app/apps-android-commons/assets/156656/9821fcff-2cb5-4c97-a4ba-cb7371b22a62

### Would you like to work on the issue?

None

**Comment by acemarco1311** [comment]
Can I work on this issue?

**Comment by acemarco1311** [comment]
I have a similar issue, but mine got automatically rotated after the edit window (auto rotate when edit image without any further rotation), I think this is rooted from this piece of code in EditActivity.kt 
`var imageRotation = 0
val rotation = imageRotation % 360
val  newRotation = rotation + 90`

@nicolas-raoul  can I work on this problem?

**Comment by acemarco1311** [comment]
Even though I cannot reproduce the exact problem (rotated immediately after the Edit Image click), the error on my testing machine is that after clicking "Edit Image" then perform no rotation (click "Save" or "Rotate" 4 times so that the picture gets back to the original position then click "Save"), the image will be rotated 90 degree. 

Originally, I thought the problem was the logic in `EditActivity.kt` that somehow triggers rotation without clicking "Rotate". However, the problem is in the implementation of `rotateImage()` function in `TransformImageImpl". The `rotateImage()` function expects 90, 180, 270 for rotation, however, when the given rotating degree = 0 (user perform no rotation), the implementation assume it should be rotated 90 degree. 

@nicolas-raoul  PR #5760 might helps, thanks a lot!

**Comment by nicolas-raoul** [comment]
@acemarco1311 Did you get a chance to check my pull request comment at https://github.com/commons-app/apps-android-commons/pull/5760#issuecomment-2188710522 ? :-)

**Comment by mnalis** [comment]
FYI, This is still a problem in Commons apps 5.1.2

https://github.com/user-attachments/assets/2d221479-ef26-4a15-a92a-7202ae9bb8e0

**Comment by Kota-Jagadeesh** [comment]
@nicolas-raoul @mnalis The issue is no more reproducible in the current version of the app. Ig we can close this issue

**Comment by mnalis** [comment]
It is still broken for me in 6.2.1 (F-droid) on Android 15.

Video:

https://github.com/user-attachments/assets/dcbc9926-140f-466a-8794-402acbd013f3

that looks at least like two bugs:
- see how picture is originally in Fossify gallery oriented to one side? And how it is also in that same orientation when shared to Commons app? But as soon as you press `Edit image` it ***automagically*** changes orientation ? It shouldn't happen, it should be shown in same orientation as when originally shown!
- And then, when I just press `Save` in that `Edit image`, without rotating or doing anything else, returned image is ***extra*** rotated. If nothing was changed, image should've been returned unchanged.

Here is original image I worked with if it is needed for test (just please don't upload it to production Commons). I expect it might have something to do with EXIF orientation header (raw image being in one position, but EXIF saying it should be rotated), and Commons app in some places following that header, and in other places ignoring it, which leads to the mess.

```
% exiftool IMG_20250208_130121.jpg | grep -i orient
Orientation                     : Rotate 90 CW
```

[demo_IMG_20250208_130121.zip](https://github.com/user-attachments/files/24999713/demo_IMG_20250208_130121.zip)

**Comment by Kota-Jagadeesh** [comment]
@mnalis Apologies for the confusion in my previous note (The issue is not reproducible on my device). I retested the issue now and it is reproducible on Android 15 (API level 35)

**Comment by Kota-Jagadeesh** [comment]
Hi @mnalis @nicolas-raoul,

I’ve found the fix for this issue and would like to request assignment to myself.

Additionally, I found another issue: when I rotate a picture by 180 degrees, save it, and then return to the upload details screen, tapping “Edit Image” again shows the picture but it’s not oriented correctly.

https://github.com/user-attachments/assets/6818512b-a95e-4ef5-86a3-b11aa9c4f7e6

**Comment by nicolas-raoul** [comment]
@acemarco1311 if you intend to continue working on this please let us know, meanwhile @Kota-Jagadeesh feel free to work on it and possibly elaborate on the existing PR if you want. Thanks all! :-)

**Comment by Kota-Jagadeesh** [comment]
@nicolas-raoul  yeah the PR #5760, was actually almost very close. that pr correctly  had identified that the else block needed `LLJTran.OPT_DEFAULTS` to fix the **"save without edit"** bug. However, it inadvertently removed `LLJTran.OPT_XFORM_ORIENTATION` from the transform call, which is still required to handle EXIF orientation correctly during the process.

That said, and fixing the transform logic alone doesn't resolve the entire problem. as i mentioned in the above commentt, that there is a second [issue](https://github.com/commons-app/apps-android-commons/issues/5711#issuecomment-3841696783) where when re-editing an image causes it to revert to the original file instead of loading the rotated version of the pic. now i have fixed both the rotation logic (retaining the correct flags) and the re-edit workflow, and will raise the PR shortly :)
